### ISSUE TITLE
The function definition for function calls does not support generics.

### ISSUE THREAD

**Issue by lvchzh** [spring-projects/spring-ai#1118]
**Bug description**
The function definition for function calls does not support generics.

**Environment**
Spring ai 1.0.0-SNAPSHOT

**Steps to reproduce**
step1:
`public class TrainSearchScheduleService
        implements Function<RequestBean<TrainSearchScheduleReq>, ResponseBean<TrainSearchScheduleResp>> {
        ....
        }
`
step2:
Runtime-based generic type erasure leads to the loss of generic information.

**Expected behavior**
Support for defining generic classes

**Comment by mxsl-gr** [comment]
generic type is not supported yet,  because the `FunctionCallback.inputTypeSchema` is generated based on `Class`, not `Type`. In theory, it could be supported if the `Class` is replaced with `Type`.

**Comment by lvchzh** [comment]
> generic type is not supported yet, because the `FunctionCallback.inputTypeSchema` is generated based on `Class`, not `Type`. In theory, it could be supported if the `Class` is replaced with `Type`.
Yes, I have noticed this through the code and submitted changes to address the issue. 
commit 71e8cee64ea7f2df7b703d8820eb40391b8cfdd2

**Comment by csterwa** [comment]
@lvchzh is resolved with commit https://github.com/spring-projects/spring-ai/commit/71e8cee64ea7f2df7b703d8820eb40391b8cfdd2?

**Comment by markpollack** [comment]
Hard to tell from the commit as it doesn't contain tests as specified in the example.  That said, we have spent efforts to remove dependencies on type utils in spring cloud function and create our own type utils that also help with kotlin support.  Will investigate the support of 

```
public class TrainSearchScheduleService implements Function<RequestBean<TrainSearchScheduleReq>, ResponseBean<TrainSearchScheduleResp>> 
```

**Comment by tzolov** [comment]
Resolved by 432954dad7f7c3a33117848a4259bc7076865c5f
